File: UserManagement.Web/Models/InvoiceDetailModel.cs
using UserManagement.Web.Helpers;
using System;
using System.Collections.Generic;

namespace UserManagement.Web.Models
{
    public class InvoiceDetailModel
    {
        public int Id { get; set; }
        public int CustomerId { get; set; }
        public decimal? Tax { get; set; }
        public decimal? Discount { get; set; }
        public decimal TotalAmount { get; set; }
        public string Remark { get; set; }
        public Constants.InvoiceStatus Status { get; set; }
        public DateTime CreatedOn { get; set; }

        public CustomerDetailModel Customer { get; set; }

        public IEnumerable<ServiceModel> Items { get; set; }
        public IEnumerable<AttachmentModel> Attachments { get; set; }
    }
}

File: UserManagement.Web/ClientApp/src/models/invoice/invoice.detail.model.ts
import { AttachmentDetailModel } from '../attachment/attachment.detail.model';
import { CustomerDetailModel } from '../customer/customer.detail.model';
import { InvoiceItemModel } from './invoice.item.model';

export class InvoiceDetailModel {
    public id: number;
    public customerName: string;
    public invoiceNumber: string;
    public tax: number;
    public discount: number;
    public amount: number;
    public totalAmount: number;
    public remark: string;
    public createdOn: string;

    public description: string; //
    public status: number; //
    public invoiceDate: string;
    public dueDate: string; //
    public poSoNumber: number; //

    public customer: CustomerDetailModel;
    public lineAmountSubTotal: number;

    public items: Array<InvoiceItemModel>;
    public attachments: Array<AttachmentDetailModel>;

    constructor() {
        this.customer = new CustomerDetailModel();
        this.items = new Array<InvoiceItemModel>();
        this.attachments = new Array<AttachmentDetailModel>();
    }
}
